checkstyle-sonar-plugin/src/main/java/org/sonar/plugins/checkstyle/CheckstyleProfileExporter.java

```diff
@@ -63,19 +67,39 @@ public void exportProfile(RulesProfile profile, Writer writer) {
             final List<ActiveRule> activeRules = profile
                     .getActiveRulesByRepository(CheckstyleConstants.REPOSITORY_KEY);
             if (activeRules != null) {
-                final Map<String, List<ActiveRule>> activeRulesByConfigKey =
-                        arrangeByConfigKey(activeRules);
+                final List<ActiveRuleWrapper> activeRuleWrappers = new ArrayList<>();
+                for (ActiveRule activeRule : activeRules) {
+                    activeRuleWrappers.add(new ActiveRuleWrapperImpl(activeRule));
+                }
+                final Map<String, List<ActiveRuleWrapper>> activeRulesByConfigKey =
+                        arrangeByConfigKey(activeRuleWrappers);
```

Comment:
Whenever you call `arrangeByConfigKey`, right before it is an array converting a list of ActiveRules into a ActiveRuleWrapper. Since both calls do the same thing this looks like it can be placed inside the `arrangeByConfigKey` method.

Reply:
No, unfortunately not. That's the issue I explained in a comment earlier that the `ActiveRule` from `RulesProfile` is server sided and a new `ActiveRule` (different package) from `ActiveRules` is scanner sided. Therefore, I convert the corresponding `ActiveRule` to the wrapper which returns the values we need in the exporter.

Reply:
Sorry, I missed the different class names in the `new`.